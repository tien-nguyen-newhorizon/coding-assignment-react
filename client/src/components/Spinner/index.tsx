const Spinner = () => {
  return (
    <div className="w-full flex justify-center items-center">
      <div className="animate-spin h-8 w-8 border-4 border-blue-500 rounded-full border-t-transparent"></div>
    </div>
  );
};

export default Spinner;
