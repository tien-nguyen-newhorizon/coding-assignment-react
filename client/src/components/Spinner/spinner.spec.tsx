import { render } from '@testing-library/react';

import Spinner from './index';

describe('Spinner', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Spinner />);
    expect(baseElement).toBeTruthy();
  });
});

describe('Spinner Snapshot', () => {
  it('should match the previous snapshot', () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
