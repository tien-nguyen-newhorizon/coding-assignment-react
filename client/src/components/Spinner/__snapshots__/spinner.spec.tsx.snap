// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Snapshot should match the previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full flex justify-center items-center"
  >
    <div
      class="animate-spin h-8 w-8 border-4 border-blue-500 rounded-full border-t-transparent"
    />
  </div>
</DocumentFragment>
`;
